tests/unit/misc_test.c

```diff
@@ -183,13 +190,102 @@ CTEST2(misc, test_ctest_assert_prints_user_msg_with_params)
    platform_close_log_stream(&stream, data->log_output);
 }
 
+/*
+ * Simple test to verify lower-level conversion macros used by size_to_str()
+ */
+CTEST2(misc, test_bytes_to_fractional_value_macros)
+{
+   // Run through a few typical cases to check lower-level macros
+   size_t size = 1.25 * KiB;
+   ASSERT_EQUAL(25, B_TO_KiB_FRACT(size));
+
+   size = 1.5 * MiB;
+   ASSERT_EQUAL(50, B_TO_MiB_FRACT(size));
+
+   size = 1.75 * GiB;
+   ASSERT_EQUAL(75, B_TO_GiB_FRACT(size));
+
+   size = 1.50 * TiB;
+   ASSERT_EQUAL(50, B_TO_TiB_FRACT(size));
+}
+
+/*
+ * Test case to verify conversion of size value to a string with
+ * unit-specifiers. Do quick cross-check of the lower-level conversion macros
+ * involved.
+ */
+CTEST2(misc, test_size_to_str)
+{
+   char size_str[SIZE_TO_STR_LEN];
+   char size_str2[SIZE_TO_STR_LEN + 2]; // size-as-string enclosed in ()
+   typedef struct size_to_expstr {
+      size_t size;
+      char  *expstr;
+   } size_to_expstr;
+
+   // Verify base-case, and validate ptr returned to output string buffer.
+   size_t size   = 0;
+   char  *expstr = "0 bytes";
+   char  *outstr = size_to_str(size_str, sizeof(size_str), size);
+   // Conversion fn should return input-str's start addr as outstr.
+   ASSERT_TRUE(outstr == size_str);
```

Comment:
Validates that the returned string `char *` is, indeed, the start addr of `outbuf`; i.e. `size_str[]`.